plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -196,6 +217,24 @@ public List<String> getAllViews(String databaseName)
                 () -> getTablesWithParameter(databaseName, PRESTO_VIEW_FLAG, "true"));
     }
 
+    @Override
+    public Optional<List<SchemaTableName>> getAllViews()
+            throws TException
+    {
+        return alternativeCall(
+                exception -> !isUnknownMethodExceptionalResponse(exception),
+                chosenGetAllViewsAlternative,
+                () -> parseTableMetadata(client.getTableMeta("*", "*", ImmutableList.of(VIRTUAL_VIEW.name()))),
+                Optional::empty);
+    }
+
+    private static Optional<List<SchemaTableName>> parseTableMetadata(List<TableMeta> tablesMetadata)
```

Comment:
nit : Can we have different method name - `getSchemaTableNames` or something like that.

Reply:
Renamed